qualtran/bloqs/arithmetic/addition.py

```diff
@@ -452,7 +452,7 @@ def build_composite_bloq(
         if self.signed:
             binary_rep = QInt(self.bitsize).to_bits(self.k)
         else:
-            binary_rep = QUInt(self.bitsize).to_bits(self.k)
+            binary_rep = QUInt(self.bitsize).to_bits(self.k % (2**self.bitsize))
```

Comment:
what's going on here? should we raise an error if `self.k` is too big in the class constructor? What about the line two lines up; should that be modified as well? Are there other locations where we currently assume `self.k` fits into `self.bitsize`?

Reply:
it's not about large $k$, it's about negative $k$. modular addition uses AddK with $-mod$ but the addition is unsigned so the call to QUInt fails validation. but adding -mod is the same as adding 2^bitsize - mod in this case ... I changed the code to explicitly apply this only for negative values and added a comment

Reply:
thanks! making these things "obvious" is very helpful